main.go

```diff
@@ -106,19 +106,24 @@ func (c *Controller) handleNewCSR(key string) error {
 	// is dependent on the actual instance, to detect that a CSR was recreated with the same name
 	klog.Infof("CSR %s added", csr.Name)
 
-	if isApproved(csr) {
-		klog.Infof("CSR %s is already approved", csr.Name)
+	parsedCSR, err := parseCSR(csr)
+	if err != nil {
+		klog.Infof("error parsing request CSR: %v", err)
 		return nil
 	}
 
-	if pending := recentlyPendingCSRs(c.indexer); pending > maxPendingCSRs {
-		klog.Infof("ignoring all CSRs as too many recent pending CSRs seen: %d", pending)
+	if !isNodeBootstrapCSR(csr) {
```

Comment:
Aren't we skipping the node serving cert CSRs entirely with this check moved up here? 
(The code path from down here - https://github.com/openshift/cluster-machine-approver/blob/master/csr_check.go#L131)

Reply:
+1, may be we should refactor `authorizeCSR()` and invoke `isNodeServingCert()` from here and from within authorizeCSR()

Reply:
> Aren't we skipping the node serving cert CSRs entirely with this check moved up here?
> (The code path from down here - https://github.com/openshift/cluster-machine-approver/blob/master/csr_check.go#L131)

this checks if this is a node bootstrappping CSR for either client or server. If it is not, skip so others approvers can react.

Reply:
For server, 
> Names contained in the CSR are checked against addresses in the corresponding node's machine status
```golang
 // Check that we have a registered node with the request name
        targetMachine, ok := findMatchingMachineFromNodeRef(nodeAsking, machines)
        if !ok {
                return fmt.Errorf("No target machine")
        }

        // SAN checks for both DNS and IPs, e.g.,
        // DNS:ip-10-0-152-205, DNS:ip-10-0-152-205.ec2.internal, IP Address:10.0.152.205, IP Address:10.0.152.205
        // All names in the request must correspond to addresses assigned to a single machine.
        for _, san := range csr.DNSNames {
                if len(san) == 0 {
                        continue
                }
                var attemptedAddresses []string
                var foundSan bool
                for _, addr := range targetMachine.Status.Addresses {
                        switch addr.Type {
                        case corev1.NodeInternalDNS, corev1.NodeExternalDNS, corev1.NodeHostName:
                                if san == addr.Address {
                                        foundSan = true
                                        break
                                } else {
                                        attemptedAddresses = append(attemptedAddresses, addr.Address)
                                }
                        default:
                        }
                }
                // The CSR requested a DNS name that did not belong to the machine
                if !foundSan {
                        return fmt.Errorf("DNS name '%s' not in machine names: %s", san, strings.Join(attemptedAddresses, " "))
                }
        }

        for _, san := range csr.IPAddresses {
                if len(san) == 0 {
                        continue
                }
                var attemptedAddresses []string
                var foundSan bool
                for _, addr := range targetMachine.Status.Addresses {
                        switch addr.Type {
                        case corev1.NodeInternalIP, corev1.NodeExternalIP:
                                if san.String() == addr.Address {
                                        foundSan = true
                                        break
                                } else {
                                        attemptedAddresses = append(attemptedAddresses, addr.Address)
                                }
                        default:
                        }
                }
                // The CSR requested an IP name that did not belong to the machine
                if !foundSan {
                        return fmt.Errorf("IP address '%s' not in machine addresses: %s", san, strings.Join(attemptedAddresses, " "))
                }
        }

```